### DISCUSSION TITLE
Apple M1 Support

### DISCUSSION THREAD

**Discussion by treysmithdev** [artisan-roaster-scope/artisan#556] [Q&A] [answered]
Hi, does artisan run on MacBooks using the new M1 processor?

**Comment by MAKOMO** [comment, marked as answer]
Dear Trey,

> On 22. Jan 2021, at 04:30, Trey Smith <notifications@github.com> wrote:
> 
> 
> Hi, does artisan run on MacBooks using the new M1 processor?

Yes it runs under Rosetta 2 emulation, but you need to ensure that your devices driver comes with M1 support. By now it was verified that the current Phidgets driver works on M1 and Artisan can successfully connect to Phidgets even via the Phidget Network Server.

Cheers,
Marko

**Reply by MAKOMO** [reply]
Well, actually Artisan is confirmed to run on M1 via the Rosetta 2 emulator build into macOS. The question is if you need a device driver to connect your hardware and if this one is available on M1.

**Reply by pbadmin-git** [reply]
I've tested and the M1 Macbook runs Artisan fine. However, no temperature info is shown. 

Have contacted Toper to see if they can assist for the driver

**Reply by nicholaswise** [reply]
Can confirm, running M1 Mac Mini with Artisan 2.4.4 - Phidgets VINT hub, 4x thermocouple device - NOTHING. Tried for a few hours to get it to show temp info with no success. PLEASE update this ASAP!

**Reply by MAKOMO** [reply]
The issue is not in the Artisan code, but in the drivers for your devices. You need to contact the corresponding companies producing those drivers (Phidgets in this case). The Artisan code is emulated using Rosetta. A native version of Artisan is not expected to be available, if ever, before next year as, again, this depends on several companies and other open-source projects porting their libraries which Artisan uses to the new platform.

**Reply by MAKOMO** [reply]
@nicholaswise: I just contacted Phidgets on this and according to their tests, the Phidget driver should work on M1. Do you have the latest Phidget driver installed? Do you see your Phidgets in the drivers Phidget Control Panel?

**Reply by nicholaswise** [reply]
@MAKOMO Thanks for contacting them! Yes I see the phidget in the control panel and I am getting a temp reading in there, but every single thing I try in Artisan does not show the temp reading when I turn it on. I installed the previous version, and that at least showed a -0.1 reading, but the most current version has a broken zero in the display. Had not seen that before

**Reply by MAKOMO** [reply]
@nicholaswise : Thanks for the quick reply. This is strange. If the driver works, Artisan should receive the readings. You made sure to terminate the Phidget Control Panel before you tried to connect with Artisan? Could you recheck? As on Intel Macs, only one of those apps can use the driver. The brocken zero are the new -1.

**Reply by nicholaswise** [reply]
@MAKOMO Yes I closed the control panel before trying to connect with Artisan. Device assignment is as follows: Phidget TMP1101 4xTC 01 - Under network: Host matches host in control panel,  no password, port 8989, remote only checked. Phidgets VINT hub is active (red light on), probes are correctly positioned, getting reading within control panel. Still broken zeros

**Reply by nicholaswise** [reply]
@MAKOMO I tried every device with different network options and nothing works! Please advice i need logging asap

**Reply by MAKOMO** [reply]
Are you using a HUB5000 for network access or what is network Phidget server? Can you successful connect with your Artisan settings from another (non M1 computer)? To me it is now unclear if your issue is a configuration issue or an issue with the M1 support of Artisan or Phidgets. I have no M1 machine to test and will not be able to afford one in the next 2-3 years. Just contacted Phidgets and they say that they tested on Apples M1 developer machine and it worked. They had to return that that machine and will get new M1 machines within the next few weeks. 

Could you share your Artisan and Phidget settings to ensure that this is not a setup problem.

I just connected Artisan to an OUT1002 module connected to my HUB5000 without any problem (on my Intel Mac).

**Reply by nicholaswise** [reply]
@MAKOMO Yes, using a 4x Thermocouple Phidget going into a HUB5000, transmitting over wifi to the phidget control panel (getting a reading there). Artisan- I have the TMP1101 4xtc 01 device selected. 
![Screen Shot 2021-03-25 at 4 32 06 PM](https://user-images.githubusercontent.com/29719079/112539792-cc24d780-8d87-11eb-9c0f-e5322f17d059.png)
![Screen Shot 2021-03-25 at 4 32 17 PM](https://user-images.githubusercontent.com/29719079/112539800-cdee9b00-8d87-11eb-97e4-0598b0994d61.png)
![Screen Shot 2021-03-25 at 4 32 30 PM](https://user-images.githubusercontent.com/29719079/112539809-cf1fc800-8d87-11eb-9ecf-181a774fd65a.png)
![Screen Shot 2021-03-25 at 4 32 45 PM](https://user-images.githubusercontent.com/29719079/112539814-d050f500-8d87-11eb-8346-37ac2095d7d5.png)

**Reply by MAKOMO** [reply]
First of all it is totally unclear why you started a Phidget Network Server on your mac? That local Phidget Network Server will serve the Phidgets connected to your Mac via USB over the network. Not needed if you only want to access the Phidget Network Server running on your HUB5000. 

That one is running on Host "hub5000.local". So put this string (without the quotes) as Host in the field of the Phidget tab in Artisan where you currently have entered "IR12". That should work.

**Reply by nicholaswise** [reply]
@MAKOMO SUCCESS! Thank you so much for your help sir! You are a lifesaver!

**Reply by CarefreeBuzzBuzz** [reply]
> @MAKOMO SUCCESS! Thank you so much for your help sir! You are a lifesaver! 

@nicholaswise I hope you will consider a generous contribution to the Artisan project given this result. 

https://artisan-scope.org/donate/

**Reply by pbadmin-git** [reply]
Apologies for the late reply, I had success with the M1 Macbook Air, Artisan ran fine using Rosetta but I needed to install the Phidgets driver and the correct TMP version. Once installed, the USB connected to COM4 ok and we have live temp data coming into artisan. No issues with M1 Mac and Artisan.

**Reply by thecrazybeardedman** [reply]
> Apologies for the late reply, I had success with the M1 Macbook Air, Artisan ran fine using Rosetta but I needed to install the Phidgets driver and the correct TMP version. Once installed, the USB connected to COM4 ok and we have live temp data coming into artisan. No issues with M1 Mac and Artisan.

Hi! how do I install the phidgets driver and the correct tmp version? I'm having a hard time using my m1 for artisan. I can't see 
<img width="1066" alt="Screenshot 2023-06-13 at 7 06 50 PM" src="https://github.com/artisan-roaster-scope/artisan/assets/136457745/a2e8ed3a-ee74-4b31-b400-dc656f0259e5">
COM4 unlike in windows. I'm using an apple USB-A to USB-C converter to connect my Kaleido M2 to my m1 mac

<img width="1107" alt="Screenshot 2023-06-13 at 7 06 32 PM" src="https://github.com/artisan-roaster-scope/artisan/assets/136457745/12bb0317-1090-4565-9ed7-465eea5573d9">
